test/e2e/apps/job.go

```diff
@@ -279,29 +279,17 @@ var _ = SIGDescribe("Job", func() {
 		job, err := e2ejob.CreateJob(ctx, f.ClientSet, f.Namespace.Name, job)
 		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
 
-		ginkgo.By("Ensuring pods aren't created for job")
-		err = framework.Gomega().Consistently(ctx, framework.HandleRetry(func(ctx context.Context) ([]v1.Pod, error) {
-			pods, err := e2ejob.GetJobPods(ctx, f.ClientSet, f.Namespace.Name, job.Name)
-			if err != nil {
-				return nil, fmt.Errorf("failed to list pod for a given job %s in namespace %s: %w", job.Name, f.Namespace.Name, err)
-			}
-			return pods.Items, nil
-		})).WithPolling(framework.Poll).WithTimeout(wait.ForeverTestTimeout).Should(gomega.BeEmpty())
-		framework.ExpectNoError(err, "failed to confirm that pods aren't created for job %s in namespace %s", job.Name, f.Namespace.Name)
-
 		ginkgo.By("Checking Job status to observe Suspended state")
-		job, err = e2ejob.GetJob(ctx, f.ClientSet, f.Namespace.Name, job.Name)
-		framework.ExpectNoError(err, "failed to retrieve latest job object")
-		exists := false
-		for _, c := range job.Status.Conditions {
-			if c.Type == batchv1.JobSuspended {
-				exists = true
-				break
-			}
-		}
-		if !exists {
-			framework.Failf("Job was expected to be completed or failed")
+		job, err = e2ejob.WaitForJobSuspend(ctx, f.ClientSet, f.Namespace.Name, job.Name)
+		framework.ExpectNoError(err, "failed to get suspend state in namespace: %s", f.Namespace.Name)
+
+		ginkgo.By("Ensuring pods aren't created for job")
+		pods, err := e2ejob.GetJobPods(ctx, f.ClientSet, f.Namespace.Name, job.Name)
+		if err != nil {
+			framework.ExpectNoError(err, "failed to list pod for a given job %s in namespace %s", job.Name, f.Namespace.Name)
```

Comment:
this looks repeated. Also, we shouldn't need `if err != nil` if we have `framework.ExpectNoError`

Reply:
updated.